✨ [feat] #28 카테고리 삭제 API 구현

yeeun0702 opened the pull request:
## 🍞 Issue
카테고리 삭제 기능을 구현했습니다. 
 Closes #28 

## 🥐 Todo
- `CategoryRemover` 클래스 생성: 카테고리 삭제를 담당하는 클래스
- `CategoryFacade`에 `deleteCategory` 메서드 추가: `CategoryService`에서 카테고리 삭제 로직을 `CategoryFacade`로 위임
- `CategoryService`에서 `deleteCategory` 메서드 호출: 비즈니스 로직을 처리하고, 카테고리 삭제를 `CategoryFacade`로 위임하여 처리

## 🧇 Details

DDD 원칙에 맞춰 영속성 객체와 도메인 객체를 분리했습니다. 또한, Facade 패턴을 통해 로직을 분리하고, 비즈니스 로직을 더 깔끔하게 관리할 수 있도록 했습니다.

- 영속성 객체 (`CategoryEntity`) 대신 도메인 객체 (`Category`)를 반환하도록 `CategoryRetriever`, `CategorySaver`, `CategoryRemover`에서 수정하여, 비즈니스 로직과 영속성 로직을 분리했습니다.
- Facade 패턴을 사용하여 카테고리 조회, 저장, 삭제 로직을 간소화하고, 각 책임을 명확하게 분리하여 유지보수성을 높였습니다.
- `CategoryService`에서 `CategoryFacade`를 통해 카테고리 삭제를 처리하도록 변경하여, 서비스에서의 책임을 간결하게 만들었습니다.

## 🖼 Postman Screenshots
<img width="1007" height="375" alt="image" src="https://github.com/user-attachments/assets/2dd0ba6d-62ab-4fdc-ab03-4a6a64ed0ba5" />


## 🍩 Reviewer Notes
이전 피알을 안 닫아서 작업하던 파일 변경이 많은 거 같은데 감안해서 봐주세용 ㅠㅠ 
그리구 DDD 원칙을 지키려구 노력했는데 잘 지켜졌을까용 ? 
또 Facade 패턴도 잘 적용되었는지 확인 부탁드려용 !!

yeeun0702 committed:
✨ [feat] #25 카테고리 수정 비즈니스 로직 구현

yeeun0702 committed:
✨ [feat] #25 카테고리 수정 API 구현

yeeun0702 committed:
♻️ [refac] #25 DTO 네이밍 변경

yeeun0702 committed:
♻️ [refac] #25 다짐 메세지 엔드포인트 변경

yeeun0702 committed:
♻️ [refac] #25 다짐 메세지 API Facade 패턴 적용

yeeun0702 committed:
✨ [feat] #25 "숨기기" -> "그만하기" 기획 변경에 따른 `isVisible` -> `isStopped` 네이밍 변경

Commit:
♻️ [refac] #25 변수명 통일 및 삼항연산자 사용

Commit:
♻️ [refac] #25 카테고리 이름 공백포함 25자로 제한

yeeun0702 committed:
♻️ [refac] #25 DDD 원칙 적용 및 Facade 패턴 도입으로 책임 분리 개선

yeeun0702 committed:
✨ [feat] #28 카테고리 삭제 API 구현

choyeongju commented on bbangzip-api/src/main/java/org/sopt/category/controller/CategoryController.java:
```diff
@@ -24,25 +28,44 @@ public class CategoryController {
         return ResponseEntity
                 .status(HttpStatus.CREATED)
-                .body(categoryService.createCategory(dummyUserId, categoryCreateRequest));
+                .body(categoryService.createCategory(dummyUserId, categoryCreateReq));
```
반영 쪼아요 ~~!

choyeongju commented on bbangzip-domain/src/main/java/org/sopt/category/domain/Category.java:
```diff
@@ -21,9 +21,20 @@ public static Category fromEntity(CategoryEntity entity) {
 
+    public Category update(String newName, CategoryColor newColor, boolean newIsStopped, int newOrder) {
+        return Category.builder()
+                .id(this.id)  // 기존 id 유지
```
this는 그냥 명시적인건가용?!

yeeun0702 replied:
네 맞아용 !!

choyeongju reviewed (approved):
너무 수고하셨습니다 ! 다 잘 갖춰진 것 같아요!!! 하다가 궁금한점 있었거나 더 나은 방향이 있었다면 공유주세요 :D!!

Commit:
🚑 [hotfix] #28 카테고리 삭제 API 추가로 인한 충돌 해결

# Conflicts:
#	bbangzip-api/src/main/java/org/sopt/category/controller/CategoryController.java
#	bbangzip-api/src/main/java/org/sopt/category/service/CategoryService.java
#	bbangzip-domain/src/main/java/org/sopt/category/facade/CategoryFacade.java
